projects/schematics/src/shared/utils/lib-utils.ts

```diff
@@ -81,282 +79,295 @@ export function shouldAddFeature(features: string[], feature: string): boolean {
-    );
-    return change;
-  }
+    if (!buildPaths.length) {
+      throw new SchematicsException(
+        "Could not find any tsconfig file. Can't find SpartacusFeaturesModule."
+      );
+    }
 
-  let configChange = `{
-    ${lazyLoadingModule}
-}
-`;
-
-  const featureModuleConfig = getConfig(
-    currentFeatureModulesConfig,
-    config.name
-  );
-  if (featureModuleConfig) {
-    const lazyLoadConfig = getConfig(featureModuleConfig, 'module');
-    if (lazyLoadConfig) {
-      return new NoopChange();
+    const basePath = process.cwd();
+    const tasks = [];
+    for (const tsconfigPath of buildPaths) {
+      tasks.push(
+        ensureModuleExists({
+          name: `${config.name}-feature`,
+          path: 'app/spartacus/features',
+          module: SPARTACUS_FEATURES_MODULE,
+          project: options.project,
+        })
+      );
+      tasks.push(addRootModule(tsconfigPath, basePath, config));
+      tasks.push(addFeatureModule(tsconfigPath, basePath, config, options));
+      tasks.push(addFeatureTranslations(tsconfigPath, basePath, config));
+      tasks.push(addCustomConfig(tsconfigPath, basePath, config));
     }
-    configChange = lazyLoadingModule;
-
-    const nestedProperty = findNodes(
-      featureModuleConfig,
-      ts.SyntaxKind.PropertyAssignment,
-      1,
-      true
-    )[0] as ts.PropertyAssignment;
-    return createNewConfig(
-      moduleSource.fileName,
-      nestedProperty,
-      config.name,
-      configChange
-    );
-  } else {
-    return createNewConfig(
-      moduleSource.fileName,
-      currentFeatureModulesConfig,
-      config.name,
-      `{
-        ${lazyLoadingModule}
-        }`
-    );
-  }
+    return chain(tasks);
+  };
 }
 
-function provideI18NConfig(
-  host: Tree,
-  source: ts.SourceFile,
-  i18nConfig: I18NConfig
-): Change[] {
-  const changes: Change[] = [];
-  if (!isImported(source, PROVIDE_CONFIG_FUNCTION, SPARTACUS_CORE)) {
-    const importChange = createImportChange(
-      host,
-      source.fileName,
-      PROVIDE_CONFIG_FUNCTION,
-      SPARTACUS_CORE
-    );
-    changes.push(importChange);
-  }
-
-  if (
-    !isImported(source, i18nConfig.resources, i18nConfig.importPath) &&
-    !isImported(source, i18nConfig.chunks, i18nConfig.importPath)
-  ) {
-    const resourceImportChange = createImportChange(
-      host,
-      source.fileName,
-      i18nConfig.resources,
-      i18nConfig.importPath
-    );
-    const chunkImportChange = createImportChange(
-      host,
-      source.fileName,
-      i18nConfig.chunks,
-      i18nConfig.importPath
-    );
+function addRootModule(
+  tsconfigPath: string,
+  basePath: string,
+  config: FeatureConfig
+) {
+  return (tree: Tree): Tree => {
+    const { appSourceFiles } = createProgram(tree, basePath, tsconfigPath);
+    appSourceFiles.forEach((sourceFile) => {
+      if (
+        sourceFile
+          .getFilePath()
+          .includes(`${dasherize(config.name)}-feature.module.ts`)
```

Comment:
Let's also do a micro optimization to dasherize the `config.name` at the beginning of the method, and not not for each file. 
This can be done for most of the occurrences of `dasherize` (if not all).